### PR TITLE
update vnstatd module

### PR DESCRIPTION
**Pull request by cgzones**

### CODE REVIEW

File: vnstatd.te [hunk 1]
```diff
@@ -34,20 +38,21 @@ allow vnstatd_t self:process signal;
 allow vnstatd_t self:fifo_file rw_fifo_file_perms;
 allow vnstatd_t self:unix_stream_socket { accept listen };
 
+manage_dirs_pattern(vnstatd_t, vnstatd_runtime_t, vnstatd_runtime_t)
+manage_files_pattern(vnstatd_t, vnstatd_runtime_t, vnstatd_runtime_t)
+files_pid_filetrans(vnstatd_t, vnstatd_runtime_t, { dir file })
+
 manage_dirs_pattern(vnstatd_t, vnstatd_var_lib_t, vnstatd_var_lib_t)
 manage_files_pattern(vnstatd_t, vnstatd_var_lib_t, vnstatd_var_lib_t)
-files_var_lib_filetrans(vnstatd_t, vnstatd_var_lib_t, { dir file })
```
Comment (pebenito) [inline vnstatd.te]: Why is this being removed?

File: vnstatd.te [hunk 2]
```diff
@@ -60,27 +65,35 @@ miscfiles_read_localization(vnstatd_t)
...
 
+# dac_override : write /var/lib/vnstat/*
+allow vnstat_t self:capability dac_override;
 allow vnstat_t self:process signal;
 allow vnstat_t self:fifo_file rw_fifo_file_perms;
 allow vnstat_t self:unix_stream_socket { accept listen };
 
 manage_dirs_pattern(vnstat_t, vnstatd_var_lib_t, vnstatd_var_lib_t)
 manage_files_pattern(vnstat_t, vnstatd_var_lib_t, vnstatd_var_lib_t)
-files_var_lib_filetrans(vnstat_t, vnstatd_var_lib_t, { dir file })
```
Comment (pebenito) [inline vnstatd.te]: Same thing here

File: vnstatd.te [hunk 3]
```diff
@@ -60,27 +65,35 @@ miscfiles_read_localization(vnstatd_t)
...
+
 domain_use_interactive_fds(vnstat_t)
 
+files_dontaudit_search_home(vnstat_t)
 files_read_etc_files(vnstat_t)
+files_search_var_lib(vnstat_t)
 
 fs_getattr_xattr_fs(vnstat_t)
 
-logging_send_syslog_msg(vnstat_t)
```
Comment (pebenito) [inline vnstatd.te]: Why is this being removed

File: vnstatd.te (hunk 1 above)
Comment (cgzones) [inline vnstatd.te]: From my point of view programs should not be able to create directories/files in /var/lib.
These should be created by the package-manager or the installations script.

File: vnstatd.te (hunk 3 above)
Comment (cgzones) [inline vnstatd.te]: vnstat_t is the interactive user application, from my experience it does not write to syslog

Comment (pebenito) [inline vnstatd.te]: Ok that makes sense.

File: vnstatd.te (hunk 1 above)
Comment (pebenito) [inline vnstatd.te]: Unfortunately there are many things I'd like higher-privilege processes to do on behalf of lesser/un-privileged domains, but it doesn't always happen.

Comment (cgzones) [inline vnstatd.te]: Are any distros not shipping the /var/lib directories?

**Pull request merged**